### PR TITLE
Promotion rule product limit improvements

### PR DESCRIPTION
**Pull request by nirnaeth**
**Description**
  Ref #3902 

Adding a further parameter `show_all` to the `select2` rule when picking products in the admin interface. 
The parameter should act only when the selection is initialized at page load and should not have an impact on the product drop-down.
Now the promotion rule section for products should properly show all products associated with the rule and not just up to a certain limit as described in #3902 .

**Screenshots**
With 8 products associated to the rule:

_Before_
Screenshot 2021-02-14 at 12 41 52

_After_
Screenshot 2021-02-14 at 12 42 14

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have added tests to cover this change (if needed)
- [x] I have attached screenshots to this PR for visual changes (if needed)

**Credits**
@aldesantis for the lead

### CODE REVIEW

**Comment by nirnaeth** [comment]
let me know if I have to squash or reformulate the commit messages

**Review by kennyadsl — COMMENTED** [review]
👋 Hey Monica! 

This looks great, thanks. 🙏 Please, can you squash commits into a single one?

**Comment by nirnaeth** [comment]
sure, on it!

**Comment by nirnaeth** [comment]
@kennyadsl do I need to do anything else?

**Comment by kennyadsl** [comment]
Nope, this is good thanks. We will wait for the next release before merging this. Going to add the right labels/milestone so it's clear for everybody. Thanks again! 🙏

File: backend/spec/controllers/spree/admin/search_controller_spec.rb
Comment (spaghetticode) [inline backend/spec/controllers/spree/admin/search_controller_spec.rb:107]: It seems to me that this `context` expectations are the same as the `context` above, only the `params` are slightly different, so it's not evident to me what difference in result the different params should produce... so I'm missing the reason for this second test 😅. Can you help me with that?

Comment (kennyadsl) [reply backend/spec/controllers/spree/admin/search_controller_spec.rb:107]: It is basically testing that, no matter which `per_page` you are requesting, when there's the `show_all` option it will return all the records, ignoring the `per_page` param. Makes sense @spaghetticode?

Comment (spaghetticode) [reply backend/spec/controllers/spree/admin/search_controller_spec.rb:107]: @kennyadsl sure... thanks!

**Review by kennyadsl — APPROVED** [review]
Thanks, Monica!

**Review by spaghetticode — APPROVED** [review]
@nirnaeth thank you 👍

**Pull request merged**
